aliasmgr/aliasmgr.go

```diff
@@ -0,0 +1,449 @@
+	var nextAlias lnwire.ShortChannelID
+
+	err := kvdb.Update(m.backend, func(tx kvdb.RwTx) error {
+		bucket, err := tx.CreateTopLevelBucket(aliasAllocBucket)
+		if err != nil {
+			return err
+		}
+
+		lastBytes := bucket.Get(lastAliasKey)
+		if lastBytes == nil {
+			// If the key does not exist, then we can write the
+			// startingAlias to it.
+			nextAlias = startingAlias
+
+			var scratch [8]byte
+			byteOrder.PutUint64(scratch[:], nextAlias.ToUint64())
+			return bucket.Put(lastAliasKey, scratch[:])
+		}
+
+		// Otherwise the key does exist so we can convert the retrieved
+		// lastAlias to a ShortChannelID and use it to assign the next
+		// ShortChannelID. This next ShortChannelID will then be
+		// persisted in the database.
+		lastScid := lnwire.NewShortChanIDFromInt(
+			byteOrder.Uint64(lastBytes),
+		)
+		nextAlias = getNextScid(lastScid)
+
+		var scratch [8]byte
+		byteOrder.PutUint64(scratch[:], nextAlias.ToUint64())
+		return bucket.Put(lastAliasKey, scratch[:])
+	}, func() {
+		nextAlias = lnwire.ShortChannelID{}
+	})
+	if err != nil {
+		return nextAlias, err
+	}
+
+	return nextAlias, nil
+}
+
+// getNextScid is a utility function that returns the next SCID for a given
+// alias SCID. The BlockHeight ranges from [10000, 262143], the TxIndex ranges
+// from [1, 16777215], and the TxPosition ranges from [1, 65535].
+func getNextScid(last lnwire.ShortChannelID) lnwire.ShortChannelID {
+	var (
+		next            lnwire.ShortChannelID
+		incrementIdx    bool
+		incrementHeight bool
+	)
+
+	// If the TxPosition is 65535, then it goes to 0 and we need to
+	// increment the TxIndex.
+	if last.TxPosition == 65535 {
+		incrementIdx = true
+	}
+
+	// If the TxIndex is 16777215 and we need to increment it, then it goes
+	// to 0 and we need to increment the BlockHeight.
+	if last.TxIndex == 16777215 && incrementIdx {
+		incrementIdx = false
+		incrementHeight = true
+	}
+
+	switch {
+	// If we increment the TxIndex, then TxPosition goes to 0.
+	case incrementIdx:
+		next.BlockHeight = last.BlockHeight
+		next.TxIndex = last.TxIndex + 1
+		next.TxPosition = 0
+
+	// If we increment the BlockHeight, then the Tx fields go to 0.
+	case incrementHeight:
+		next.BlockHeight = last.BlockHeight + 1
+		next.TxIndex = 0
+		next.TxPosition = 0
+
+	// Otherwise, we only need to increment the TxPosition.
+	default:
+		next.BlockHeight = last.BlockHeight
+		next.TxIndex = last.TxIndex
+		next.TxPosition = last.TxPosition + 1
+	}
+
+	return next
+}
+
+// IsAlias returns true if the passed SCID is an alias. The function determines
+// this by looking at the BlockHeight. If the BlockHeight is greater than 10000
+// and less than 2^18, then it is an alias assigned by RequestAlias. These
+// bounds only apply to aliases we generate. Our peers are free to use any
+// range they choose.
+func IsAlias(scid lnwire.ShortChannelID) bool {
+	return scid.BlockHeight >= 10000 && scid.BlockHeight < 1<<18
```

Comment:
In signet, true even if `scid` is confirmed SCID(currently block height=89000~).

Reply:
Thanks